packages/modules/solaredge/external_inverter.py

```diff
@@ -7,35 +8,54 @@
 from modules.common.component_type import ComponentDescriptor
 from modules.common.fault_state import ComponentInfo
 from modules.common.modbus import ModbusDataType
-from modules.common.simcount import SimCounter
 from modules.common.store import get_inverter_value_store
 from modules.solaredge.config import SolaredgeExternalInverterSetup
+from modules.solaredge.scale import scale_registers
+from modules.solaredge.meter import SolaredgeMeterRegisters
+
+log = logging.getLogger(__name__)
 
 
 class SolaredgeExternalInverter:
     def __init__(self,
                  device_id: int,
                  component_config: Union[Dict, SolaredgeExternalInverterSetup],
                  tcp_client: modbus.ModbusTcpClient_) -> None:
-        self.__device_id = device_id
         self.component_config = dataclass_from_dict(SolaredgeExternalInverterSetup, component_config)
         self.__tcp_client = tcp_client
-        self.__sim_counter = SimCounter(self.__device_id, self.component_config.id, prefix="pv")
+        self.registers = SolaredgeMeterRegisters(self.component_config.configuration.meter_id)
         self.__store = get_inverter_value_store(self.component_config.id)
         self.component_info = ComponentInfo.from_component_config(self.component_config)
 
     def update(self, state: InverterState) -> None:
         self.__store.set(state)
 
     def read_state(self) -> InverterState:
-        unit = self.component_config.configuration.modbus_id
-        # 40380 = "Meter 2/Total Real Power (sum of active phases)" (Watt)
-        power = self.__tcp_client.read_holding_registers(40380, ModbusDataType.INT_16, unit=unit)
-        _, exported = self.__sim_counter.sim_count(power)
+        def read_scaled_int16(address: int, count: int):
+            return scale_registers(
+                self.__tcp_client.read_holding_registers(
+                    address,
+                    [ModbusDataType.INT_16] * (count + 1),
+                    unit=self.component_config.configuration.modbus_id)
+            )
+
+        def read_scaled_uint32(address: int, count: int):
+            return scale_registers(
+                self.__tcp_client.read_holding_registers(
+                    address,
+                    [ModbusDataType.UINT_32] * count + [ModbusDataType.INT_16],
+                    unit=self.component_config.configuration.modbus_id)
+            )
```

Comment:
Die beiden Funktionen sind sehr ähnlich. Man könnte diese so zusammenfassen:
```python
def read_scaled(type: ModbusDataType, address: int, count: int):
    return scale_registers(
        self.__tcp_client.read_holding_registers(
            address,
            [type] * count + [ModbusDataType.INT_16],
            unit=self.component_config.configuration.modbus_id)
    )

read_scaled_int16 = functools.partial(read_scaled, ModbusDataType.INT_16)
read_scaled_uint32 = functools.partial(read_scaled, ModbusDataType.UINT_32)
```
Die Funktion ist allerdings auch noch doppelt in der `counter.py`. Ggf. könnte man zum Beispiel folgende Funktion in der `scale.py` ergänzen:
```python
def create_scaled_reader(client: modbus.ModbusTcpClient_, modbus_id: int, type: ModbusDataType):
    def scaled_reader(address: int, count: int):
        return scale_registers(
            client.read_holding_registers(address, [type] * count + [ModbusDataType.INT_16], unit=modbus_id)
        )
    
    return scaled_reader
```
Und dann geht in den beiden Dateien:
```python
read_scaled_int16 = create_scaled_reader(
    self.__tcp_client, self.component_config.configuration.modbus_id, ModbusDataType.INT_16
)
read_scaled_uint32 = create_scaled_reader(
    self.__tcp_client, self.component_config.configuration.modbus_id, ModbusDataType.UINT_32
)
```